client/my-sites/domains/domain-management/dataviews/domains-dataviews.tsx

```diff
@@ -0,0 +1,62 @@
+import { DataViews, filterSortAndPaginate } from '@wordpress/dataviews';
+import { useI18n } from '@wordpress/react-i18n';
+import { useMemo, useState } from 'react';
+import { BulkUpdateNotice } from './bulk-update-notice';
+import { useActions } from './use-actions';
+import { DomainData, getDomainId } from './use-domains';
+import './style.scss';
+import { useFields } from './use-fields';
+import useView from './use-view';
+
+type Props = {
+	domains: DomainData[];
+	isLoading: boolean;
+	sidebarMode?: boolean;
+	selectedDomainName?: string;
+	openDomainPane?: ( domain: DomainData ) => void;
+};
+
+export const DomainsDataViews = ( {
+	domains,
+	isLoading,
+	sidebarMode,
+	selectedDomainName,
+	openDomainPane,
+}: Props ) => {
+	const { __ } = useI18n();
+	const { view, setView } = useView( { sidebarMode } );
+
+	const fields = useFields( { openDomainPane } );
+	const actions = useActions( { sidebarMode } );
+
+	const { data: processedDomains, paginationInfo } = useMemo( () => {
+		return filterSortAndPaginate( domains, view, fields );
+	}, [ domains, view, fields ] );
+
+	const [ selectedIds, setSelectedIds ] = useState< string[] >( [] );
+	const selectedDomain = domains
+		.filter( ( d: DomainData ) => d.processed.domain === selectedDomainName )
+		.pop();
+
+	const layout = sidebarMode ? { list: {} } : { table: {} };
+	return (
+		<div className="domains-dataviews">
```

Comment:
Can dataviews be updated to the latest version in dotcom? All of this would be simplified by using the latest (see `onClickItem`/`isItemClickable` in [the docs](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-dataviews/#properties)).

Reply:
Yes, updating the library could make even more things easier. 
But I'm afraid it's not safe since the library is used in other parts of the app.
Shouldn't it be a separate task? Maybe even outside the domain-related project?